messaging/src/main/java/org/axonframework/messaging/deadletter/GenericCause.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.deadletter;
+
+import java.util.Objects;
+
+/**
+ * Generic implementation of a {@link Cause}.
+ *
+ * @author Steven van Beelen
+ * @author Mitchel Herrijgers
+ * @since 4.6.0
+ */
+class GenericCause implements Cause {
```

Comment:
Suggest using an `ExceptionCause` when the DeadLetter cause is an exception. That way, we can keep a reference to the entire exception and stack trace.

Reply:
Good point. Renamed to `ThrowableCause` though, as it consumes a `Throwable`.